packages/react-magma-dom/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,199 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { css } from '@emotion/core';
+
+import { ButtonColor, ButtonSize } from '../Button';
+import { ButtonGroup } from '../ButtonGroup';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { ToggleButton, ToggleButtonProps } from '../ToggleButton/ToggleButton';
+import { transparentize } from 'polished';
+
+/**
+ * @children required
+ */
+export interface ToggleButtonGroupProps {
+  isInverse?: boolean;
+  /**
+   * Removes margins between buttons and applies a uniform border around the group.
+   */
+  noSpace?: boolean;
+  /**
+   * Enables a radio configuration throughout the group retaining an active selection at all times.
+   */
+
+  requiredSelect?: boolean;
+  /**
+   * Sets the Toggle Button group to have only one active selection.
+   */
+  singleSelect?: boolean;
+  /**
+   * Changes the button sizes throughout the group between 'small', 'medium', and 'large'.
+   */
+  size?: ButtonSize;
+  /**
+   * @internal
+   */
+  testId?: string;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+
+export interface ToggleButtonGroupContextInterface {
+  isDefault?: boolean;
+  isInverse?: boolean;
+  requiredSelect?: boolean;
+  singleSelect?: boolean;
+  size?: ButtonSize;
+}
+
+export const ToggleButtonGroupContext =
+  React.createContext<ToggleButtonGroupContextInterface>({});
+
+//Sets the icon width for icon only Toggle Buttons.
+export function setIconWidth(props: ToggleButtonProps) {
+  if (props.size === ButtonSize.small) {
+    return props.theme.spaceScale.spacing07;
+  }
+  if (props.size === ButtonSize.medium) {
+    return props.theme.spaceScale.spacing09;
+  }
+  if (props.size === ButtonSize.large) {
+    return props.theme.spaceScale.spacing11;
+  }
+  return props.theme.spaceScale.spacing09;
+}
+
+//Sets the background colors for the Toggle Buttons.
+function setBackgroundColor(props) {
+  //Removes active background color.
+  if (
+    (props.isChecked && props.defaultChecked) ||
+    (!props.isChecked && props.isDefault)
+  ) {
+    return '';
+  }
+  //Sets active background color.
+  if (props.isChecked || props.defaultChecked || props.isDefault) {
+    if (props.isInverse) {
+      return transparentize(0.5, props.theme.colors.neutral900);
+    }
+    return transparentize(0.5, props.theme.colors.neutral300);
+  }
+}
+
+export const setButtonStyles = props => css`
```

Comment:
Because these are Button styles, should they live in ToggleButton and then get imported here in ToggleGroup?